modules/ftp/scanner.go

```diff
@@ -279,7 +279,7 @@ func (s *Scanner) Scan(t zgrab2.ScanTarget) (status zgrab2.ScanStatus, result in
 	}
 	if s.config.FTPAuthTLS && is200Banner {
 		if err := ftp.GetFTPSCertificates(); err != nil {
-			return zgrab2.SCAN_APPLICATION_ERROR, &ftp.results, err
+			return zgrab2.TryGetScanStatus(err), &ftp.results, err
```

Comment:
Let's wrap the errors returned in error checking blocks throughout the `Scan` function so we can disambiguate where the error originated during debugging.